pkg/source/control/message_header.go

```diff
@@ -0,0 +1,75 @@
+package control
+
+import (
+	"encoding/binary"
+	"io"
+
+	"github.com/google/uuid"
+)
+
+type MessageFlag uint8
+
+type MessageHeader struct {
```

Comment:
can we add some documentatin on the Message format etc once this is out of POC state ?

Reply:
I've evolved that part in the prototype, I still need to sync here: https://github.com/slinkydeveloper/control-data-plane-communication/blob/master/pkg/control/message.go#L17